src/styles/buttons.less

```diff
@@ -1179,4 +1179,272 @@ fieldset[disabled] .btn-toggle.active {
       color: @thumbnail-icon-color;
     }
   }
+}
+
+// focus on tab;
+
+.button-keyboard-focus:not(.disabled):not([disabled]) {
+  &.btn {
```

Comment:
Please explain this.

Reply:
For this css only approach the button itself does not contain the styling, we then add a span inside it which is styled like the buttons and given a tab index of -1, this is why styling has been repeated for the most part. This allows us to set up styling so that on click we will will not trigger any focus styles so focus will only occur when using keyboard controls.

Reply:
Can't you add this to the existing button style?

Reply:
The existing styles will be styling the .btn class, these styles are overwriting those styles and styling the .button_container class instead so there wont be any overlap. I thought keeping them separate from the existing buttons was tidier and easier to read but I can move them if you think thats preferable.

Reply:
What differences are there? It looks to be mostly copy pasted

Reply:
Okay, I'm wrong on that, don't worry about it.